tests/crawler/test_crawler_unittest.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from selenium import webdriver
+from webdriver_manager.firefox import GeckoDriverManager
+
+from coverage_crawler import crawler
+
+
+class TestCrawler(unittest.TestCase):
+    driver: webdriver.Firefox
+
+    def setUp(self):
+        self.driver = webdriver.Firefox(executable_path=GeckoDriverManager().install())
+
+    def tearDown(self):
+        self.driver.close()
+
+    def test_close_all_windows_except_first(self):
+        """
+        when multiple windows are open,
+        verify that calling `close_all_windows_except_first` closes all except the first. 
+        """
+        _open_google_tab_script = 'window.open("https://google.com", "new window")'
```

Comment:
Let's open an empty page, so we don't have to load a resource from the internet

Reply:
fixed